app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -287,3 +289,27 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         adapter.removeTileFromAdapter(tileId)
     }
 }
+
+/**
+ * A [ScrollView] with functionality overridden for the specific requirements of the overlay.
+ *
+ * One crappy thing with the current implementation is that when a scroll is interrupted (e.g. user
+ * clicks up twice quickly), it will skip and not scroll smoothly. Since we don't scroll often,
+ * this seems fine.
+ */
+class BrowserNavigationOverlayScrollView(
+        context: Context, attrs: AttributeSet
+) : ScrollView(context, attrs) {
+
+    private val deltaScrollPadding = resources.getDimensionPixelSize(R.dimen.browser_overlay_delta_scroll_padding)
+
+    override fun computeScrollDeltaToGetChildRectOnScreen(rect: Rect?): Int {
```

Comment:
There's already some scroll code for the RecyclerView, to autoscroll so the next row is showing - how does this interact with that? Should that be deleted?